cms/djangoapps/contentstore/views/tests/test_transcript_settings.py

```diff
@@ -0,0 +1,167 @@
+import ddt
+import json
+from mock import Mock, patch
+
+from django.test.testcases import TestCase
+
+from contentstore.tests.utils import CourseTestCase
+from contentstore.utils import reverse_course_url
+from contentstore.views.transcript_settings import validate_transcript_credentials
+
+
+@ddt.ddt
+@patch(
+    'openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled',
+    Mock(return_value=True)
+)
+class TranscriptCredentialsTest(CourseTestCase):
+    """
+    Tests for transcript credentials handler.
+    """
+    VIEW_NAME = 'transcript_credentials_handler'
+
+    def get_url_for_course_key(self, course_id):
+        return reverse_course_url(self.VIEW_NAME, course_id)
+
+    def test_302_with_anonymous_user(self):
+        """
+        Verify that redirection happens in case of unauthorized request.
+        """
+        self.client.logout()
```

Comment:
A comment to mention why we need the logout for future references.

Reply:
so that we can make unauthorized request -- described in docstring.